src/main/java/jenkins/plugins/http_request/HttpRequest.java

```diff
@@ -58,46 +61,79 @@
  */
 public class HttpRequest extends Builder {
 
-    private final String url;
-    private HttpMode httpMode;
-    private MimeType contentType;
-    private MimeType acceptType;
-    private final String outputFile;
-    private final String authentication;
-    private Boolean consoleLogResponseBody;
-    private Boolean passBuildParameters;
-    private List<NameValuePair> customHeaders = new ArrayList<NameValuePair>();
-    private final Integer timeout;
-    private String validResponseCodes;
-    private String validResponseContent;
-
-    /**
-     * @deprecated only to deserialize and serialize in a new form
-     */
-    @Deprecated
-    private Boolean returnCodeBuildRelevant;
+    private @Nonnull String url;
+    private HttpMode httpMode                 = DescriptorImpl.httpMode;
+    private Boolean passBuildParameters       = DescriptorImpl.passBuildParameters;
+    private String validResponseCodes         = DescriptorImpl.validResponseCodes;
+    private String validResponseContent       = DescriptorImpl.validResponseContent;
+    private MimeType acceptType               = DescriptorImpl.acceptType;
+    private MimeType contentType              = DescriptorImpl.contentType;
+    private String outputFile                 = DescriptorImpl.outputFile;
+    private Integer timeout                   = DescriptorImpl.timeout;
+    private Boolean consoleLogResponseBody    = DescriptorImpl.consoleLogResponseBody;
+    private String authentication             = DescriptorImpl.authentication;
```

Comment:
Is this really supposed to be stored in plaintext? Normally fields like this would at least be `Secret`s (if not the `id` of some subclass of `Credentials`).